docs/BRANCHING.md

```diff
@@ -0,0 +1,51 @@
+# Branching Strategy for releases
+
+## General Considerations
+
+- Generally the master branch is the one where current development is happening.
+- One objective would be to arrive to a point where the master branch can be released very often (every day?).
+  This means the master branch is releasable at all times. I believe this strategy is called **Continuous Delivery**.
+- At this point we don't think it's possible to do that until the testing infrastructure has been improved.
+- Nevertheless, the branching strategy takes this objective into account.
+- To further support this objective, the strategy wil try to minimize the number of long-lasting branches.
+
+## Release numbering
+- example 4.3.0
+- The first number (4 in this example) is incremented for major changes in the software. What major change mean has to be decided.
+- The second number (3 in this example) is incremented for minor changes in the software.
+- The third number (0 in this example) is used for maintenance releases.
+  A maintenance is created if some bug has to be corrected before the next minor release.
```

Comment:
@jcpitre Can we link to an example? Such as this one: https://github.com/MobilityData/gtfs-validator/pull/1653

Reply:
We could, except we decided not to do a maintenance release with the PR you mentioned since there was a workaround that did not involve a release. So the example is might not be as convincing as we want.

Reply:
Frankly I can't find any. The best is the one you mentioned. So I can put it in the document and add a comment in #1663 saying that this should lead to a maintenance release but we were lucky to have a workaround.

Reply:
Added the link